libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthenticationFactory.cs

```diff
@@ -68,6 +69,23 @@ public static class BotFrameworkAuthenticationFactory
                     httpClient,
                     logger);
             }
+            else
+            {
+                // else apply the built in default behavior, which is either the public cloud or the gov cloud depending on whether we have a channelService value present 
+
+                if (string.IsNullOrEmpty(channelService))
+                {
+                    return new PublicCloudBotFrameworkAuthentication(credentialFactory, authConfiguration, httpClient, logger);
+                }
+                else if (channelService == GovernmentAuthenticationConstants.ChannelService)
+                {
+                    return new GovernmentCloudBotFrameworkAuthentication(credentialFactory, authConfiguration, httpClient, logger);
+                }
+                else
+                {
+                    throw new Exception("A ChannelService was given but the value was not recognized.");
```

Comment:
[`throw new ArgumentException`](https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception?view=netcore-3.1) instead of `throw new Exception`?

`"The provided ChannelService value is not supported by BotFrameworkAuthenticationFactory.Create(). "`

Then perhaps something like _`"Implement a subclass of BotFrameworkAuthentication or provide a recognized ChannelService value."`_ as the remaining exception message or as a comment in the code?

Reply:
nice catch!